### PR TITLE
wip start of visit create process, retrieval of visit from VSIP on re…

### PR DESCRIPTION
**Pull request by lbennett-moj**
…ceipt of message

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/listeners/PrisonerDomainEventsListener.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ class PrisonerDomainEventsListener(
     val changeEvent: HMPPSDomainEvent = gson.fromJson(sqsMessage.Message, HMPPSDomainEvent::class.java)
     log.info("Received ${changeEvent.eventType} event")
     when (changeEvent.eventType) {
-      "prison-visit.booked" -> prisonVisitsService.createVisit()
+      "prison-visit.booked" -> prisonVisitsService.createVisit(changeEvent.visitId)
       "prison-visit.revised" -> prisonVisitsService.updateVisit()
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/listeners/PrisonerDomainEventsListener.kt:28]: id to be retrieved from the additional details section

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+
+@Service
+class NomisApiService(@Qualifier("nomisApiWebClient") private val webClient: WebClient) {
+
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:11]: this service is wip and subject to change

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt [hunk 3]
```diff
@@ -20,7 +20,7 @@ fun prisonVisitCreatedMessage() = """
         "MessageId": "48e8a79a-0f43-4338-bbd4-b0d745f1f8ec", 
         "Token": null, 
         "TopicArn": "arn:aws:sns:eu-west-2:000000000000:hmpps-domain-events", 
-        "Message": "{\"eventType\":\"prison-visit.booked\"}", 
+        "Message": "{\"eventType\":\"prison-visit.booked\", \"visitId\":\"12\"}", 
         "SubscribeURL": null,
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt:24]: visit Id will move to additional details section

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/VisitsApiMockServer.kt [hunk 4]
```diff
@@ -32,6 +34,8 @@ class VisitsApiMockServer : WireMockServer(WIREMOCK_PORT) {
     private const val WIREMOCK_PORT = 8082
   }
 
+  private val mapper = ObjectMapper()
+
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/VisitsApiMockServer.kt:38]: mapper will go

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**